rsql-jpa/src/main/java/io/github/perplexhub/rsql/SortUtils.java

```diff
@@ -31,26 +29,20 @@ static List<Order> parseSort(@Nullable final String sort, final Map<String, Stri
             .collect(Collectors.toList());
     }
 
+    @SuppressWarnings("unchecked")
     private static Order sortToJpaOrder(final String[] parts, final Map<String, String> propertyMapper, final Root<?> root, final CriteriaBuilder cb) {
         final String property = parts[0];
         final String direction = parts[1];
 
-        final String propertyPath = propertyMapper.getOrDefault(property, property);
-        Expression<?> propertyExpression = pathToExpression(root, propertyPath);
+        final RSQLJPAPredicateConverter rsqljpaPredicateConverter = new RSQLJPAPredicateConverter(cb, propertyMapper);
+        final RSQLJPAContext rsqljpaContext = rsqljpaPredicateConverter.findPropertyPath(property, root);
+
+        Expression<?> propertyExpression = rsqljpaContext.getPath();
```

Comment:
Interesting.

Reply:
In this case `RSQLJPAPredicateConverter` has already built-in functionality to build correct `Path` for given `propertyPath`. Also there is already correct support for `join`. That's why it make sense to reuse it